app/prepare_proposal.go

```diff
@@ -12,7 +12,8 @@ import (
 // preparing the proposal block data. The square size is determined by first
 // estimating it via the size of the passed block data. Then, this method
 // generates the data root for the proposal block and passes it back to
-// tendermint via the BlockData.
+// tendermint via the BlockData. Panics indicate a developer error and should
+// immediately halt the node for visibility and so they can be quickly resolved.
```

Comment:
are these docs up to date with the below?

>// In the case of a panic from an unexpected condition, it is better for the liveness of the
	// network that we catch it, log an error and vote nil than to crash the node.

Reply:
Yeah. `PrepareProposal` has panics, `ProcessProposal` catches them.